crates/router/src/core/refunds.rs

```diff
@@ -75,7 +81,7 @@ pub async fn refund_create_core(
     })?;
 
     payment_attempt = db
-        .find_payment_attempt_last_successful_attempt_by_payment_id_merchant_id(
```

Comment:
you can refactor the existing method to accept a vector of payment statuses, you can create an issue for this and take up in a separate PR